File: litho-it/src/test/java/com/facebook/litho/GetSimpleNameTest.java
/*
 * Copyright 2014-present Facebook, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.facebook.litho;

import static org.assertj.core.api.Java6Assertions.assertThat;

import com.facebook.litho.testing.TestDrawableComponent;
import com.facebook.litho.testing.testrunner.ComponentsTestRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RuntimeEnvironment;

@RunWith(ComponentsTestRunner.class)
public class GetSimpleNameTest {

  private static class TestWrapperComponent extends Component {

    private final Component mDelegate;

    protected TestWrapperComponent(Component delegate) {
      super("TestWrapper");
      mDelegate = delegate;
    }

    @Override
    protected Component getSimpleNameDelegate() {
      return mDelegate;
    }
  }

  private ComponentContext mContext;

  @Before
  public void setUp() {
    mContext = new ComponentContext(RuntimeEnvironment.application);
  }

  @Test
  public void testGetSimpleName() {
    TestDrawableComponent testComponent = TestDrawableComponent.create(mContext).build();
    assertThat(testComponent.getSimpleName()).isEqualTo("TestDrawableComponent");
  }

  @Test
  public void testGetSimpleNameWithOneWrapper() {
    TestDrawableComponent inner = TestDrawableComponent.create(mContext).build();
    TestWrapperComponent wrapper = new TestWrapperComponent(inner);
    assertThat(wrapper.getSimpleName()).isEqualTo("TestWrapper(TestDrawableComponent)");
  }

  @Test
  public void testGetSimpleNameWithMultipleWrapper() {
    TestDrawableComponent inner = TestDrawableComponent.create(mContext).build();
    TestWrapperComponent wrapper = new TestWrapperComponent(inner);
    TestWrapperComponent wrapper2 = new TestWrapperComponent(wrapper);
    TestWrapperComponent wrapper3 = new TestWrapperComponent(wrapper2);
    assertThat(wrapper3.getSimpleName()).isEqualTo("TestWrapper(TestDrawableComponent)");
  }
}

File: litho-it/src/test/com/facebook/litho/GetSimpleNameTest.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.facebook.litho

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import com.facebook.litho.GetSimpleNameTest.TestWrapperComponent
import com.facebook.litho.testing.testrunner.LithoTestRunner
import com.facebook.litho.widget.SimpleMountSpecTester
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(LithoTestRunner::class)
class GetSimpleNameTest {

  private class TestWrapperComponent(private val delegate: Component) :
      SpecGeneratedComponent("TestWrapper") {
    override fun getSimpleNameDelegate(): Component = delegate
  }

  private lateinit var context: ComponentContext

  @Before
  fun setUp() {
    context = ComponentContext(ApplicationProvider.getApplicationContext<Context>())
  }

  @Test
  fun testGetSimpleName() {
    val testComponent = SimpleMountSpecTester.create(context).build()
    assertThat(testComponent.simpleName).isEqualTo("SimpleMountSpecTester")
  }

  @Test
  fun testGetSimpleNameWithOneWrapper() {
    val inner = SimpleMountSpecTester.create(context).build()
    val wrapper = TestWrapperComponent(inner)
    assertThat(wrapper.simpleName).isEqualTo("TestWrapper(SimpleMountSpecTester)")
  }

  @Test
  fun testGetSimpleNameWithMultipleWrapper() {
    val inner = SimpleMountSpecTester.create(context).build()
    val wrapper = TestWrapperComponent(inner)
    val wrapper2 = TestWrapperComponent(wrapper)
    val wrapper3 = TestWrapperComponent(wrapper2)
    assertThat(wrapper3.simpleName).isEqualTo("TestWrapper(SimpleMountSpecTester)")
  }
}
